### PR TITLE
docs: add configuration and training guide (#156)

### PR DESCRIPTION
**Pull request by omarfathy7**
## Summary

part 4 of #156

covers yaml config structure, `_target_` instantiation, data splitting, running training, and output artifacts

## AI usage

- [ ] No AI tools were used for this contribution
- [x] AI tools were used to assist this contribution

**Tool(s) and usage:**

chatgpt and claude for doc review and wording suggestions

## Testing

- [ ] Tests added or updated where needed
- [x] Existing tests and checks pass
- [x] Documentation updated if needed

### CODE REVIEW

File: docs/training.md [hunk 1]
```diff
@@ -1 +1,136 @@
...
+Here is a stripped-down config to show the overall shape:
+
+```yaml
+project_name: my_experiment
+
+preprocessor:
+  _target_: preprocessors.char_tokenizer.CharTokenizer
+
+model:
+  _target_: models.transformer.TransformerModel
```
Comment (TahaZahid05) [inline docs/training.md]: this is outdated. there is no model named `TransformerModel`

File: docs/training.md [hunk 2]
```diff
@@ -1 +1,136 @@
...
+  hidden_size: 64
+  num_layers: 2
+
+optimizer:
+  _target_: torch.optim.AdamW
+  lr: 0.0003
+
+data:
+  dataset:
+    _target_: datasets.my_dataset.MyDataset
```
Comment (TahaZahid05) [inline docs/training.md]: in generated project folder is `dataset_impl/` rather than `datasets` so target will also need to be `dataset_impl.string_reverse.StringReverseDataset`

It helps to give user example of one of the already built components in example so they can map to their usecase rather than an abstract one.

File: docs/training.md [hunk 3]
```diff
@@ -1 +1,136 @@
...
+  dataloader:
+    batch_size: 128
+    shuffle: true
+  val_ratio: 0.1
+  test_ratio: 0.1
```
Comment (TahaZahid05) [inline docs/training.md:47]: add transform as well here.

File: docs/training.md (hunk 1 above)
Comment (TahaZahid05) [inline docs/training.md]: also mention `collate_fn_target`

File: docs/training.md [hunk 4]
```diff
@@ -1 +1,136 @@
...
+```bash
+uv run tensorboard --logdir outputs
+```
+
+## Examples
+
+Working examples are the best way to understand how everything fits together:
+
+* [**String Reversal**](https://github.com/pytorch-ignite/trainite/tree/main/examples/string_reversal) — a sequence-to-sequence toy task that trains a decoder-only Transformer to reverse character strings.
+* [**Counting**](https://github.com/pytorch-ignite/trainite/tree/main/examples/counting) — a counting task using ClearML for logging.
```
Comment (TahaZahid05) [inline docs/training.md]: Rewrite a better description of counting after reading the example.

File: docs/training.md [hunk 5]
```diff
@@ -1 +1,136 @@
...
+```yaml
+optimizer:
+  _target_: torch.optim.AdamW
+  lr: 0.0003
+```
+
+This tells Trainite to instantiate `torch.optim.AdamW` with the configured learning rate. The remaining arguments (like model parameters) are injected by the trainer.
+
+The `preprocessor`, `model`, `optimizer`, and `data.dataset` blocks all use this pattern.
+
```
[the remark this answers is not in the document]
Comment (omarfathy7) [reply docs/training.md]: applied in 27f8ec7 and restored the dataset block in 232df2d after an accidental edit

File: docs/training.md (hunk 1 above)
Comment (omarfathy7) [reply docs/training.md]: addressed the example now uses models.rope_transformer.RoPETransformerModel, the default model generated by trainite init

File: docs/training.md (hunk 2 above)
Comment (omarfathy7) [reply docs/training.md]: addressed the _target_ now points to dataset_impl.string_reverse.StringReverseDataset inside the same data.dataset block

File: docs/training.md (hunk 3 above)
Comment (omarfathy7) [reply docs/training.md:47]: added a transform block to the example using dataset_impl.string_reverse.PromptCompletionTransform with ignore_index: -100

File: docs/training.md (hunk 1 above)
Comment (omarfathy7) [reply docs/training.md]: addressed collate_fn_target is now in the example and explained in the model block

File: docs/training.md (hunk 4 above)
Comment (omarfathy7) [reply docs/training.md]: rewritten the description now covers the actual task (c-rasp length generalization experiments) instead of the logger

**Comment by omarfathy7** [comment]
all review feedback addressed and pushed and the branch is up to date with main
please take another look when you have time and if anything still looks off let me know and i will fix it right away
thank you

File: docs/training.md [hunk 6]
```diff
@@ -1 +1,148 @@
...
+```bash
+uv run tensorboard --logdir outputs
+```
+
+## Examples
+
+Working examples are the best way to understand how everything fits together:
+
+* [**String Reversal**](https://github.com/pytorch-ignite/trainite/tree/main/examples/string_reversal) — a sequence-to-sequence toy task that trains a decoder-only Transformer with RoPE to reverse character strings.
+* [**Counting**](https://github.com/pytorch-ignite/trainite/tree/main/examples/counting) — reproduces length-generalization experiments from *"Knee-Deep in C-RASP: A Transformer Depth Hierarchy"* on an alternating block language, featuring multi-depth sweeps and heatmap visualization.
```
Comment (TahaZahid05) [inline docs/training.md]: still sounds too complex. explain the gist of task simply like "train a decoder-only Transformer to count through an alternating block language by reproducing "paper name"

Comment (omarfathy7) [reply docs/training.md]: done as suggested

**Review by TahaZahid05 — APPROVED** [review]

**Comment by TahaZahid05** [comment]
@omarfathy7 Thanks!

**Comment by omarfathy7** [comment]
@TahaZahid05 thx for the review guidance do u recommend any other issues? looks like the catreloaded ones r already taken and im not sure if im allowed to work on issues outside catreloaded but id be happy to help with any other open issues if possible sorry if this isnt the right place to ask

**Comment by TahaZahid05** [comment]
@omarfathy7 no worries! You can look at #125 as your next issue. It can be quite useful to Trainite.
